lapack/testlapack/dlag2.go

```diff
@@ -0,0 +1,96 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"golang.org/x/exp/rand"
+)
+
+type Dlag2er interface {
+	// DLAG2 computes the eigenvalues of a 2 x 2 generalized eigenvalue
+	// problem
+	//  A - w B,
+	// with scaling as necessary to avoid over-/underflow.
+	// The scaling factor "s" results in a modified eigenvalue equation
+	//  s A - w B
+	// where  s  is a non-negative scaling factor chosen so that  w,  w B,
+	// and  s A  do not overflow and, if possible, do not underflow, either.
+	//
+	// B is a  ldb, 2)
+	// On entry, the 2 x 2 upper triangular matrix B.  It is
+	// assumed that the one-norm of B is less than 1/SAFMIN.  The
+	// diagonals should be at least sqrt(SAFMIN) times the largest
+	// element of B (in absolute value); if a diagonal is smaller
+	// than that, then  +/- sqrt(SAFMIN) will be used instead of
+	// that diagonal.
+	//
+	// Dlag2 is an internal routine. It is exported for testing purposes.
+	Dlag2(a []float64, lda int, b []float64, ldb int) (scale1, scale2, wr1, wr2, wi float64)
+}
+
+func Dlag2Test(t *testing.T, impl Dlag2er) {
+	rnd := rand.New(rand.NewSource(1))
+	for _, lda := range []int{2, 3, 4, 5, 6, 7, 30} {
+		for _, ldb := range []int{2, 3, 4, 5, 6, 7, 30} {
+			dlag2Test(t, impl, rnd, lda, ldb)
+		}
+	}
+}
+
+func dlag2Test(t *testing.T, impl Dlag2er, rnd *rand.Rand, lda, ldb int) {
+	const tol = 1e-14
+	a := randomGeneral(2, 2, lda, rnd)
+	b := randomGeneral(2, 2, ldb, rnd)
+	b.Data[ldb] = 0 // Dlag2 requires b to be upper triangular
```

Comment:
There is no requirement in the code that this element is set to zero, it's simply not referenced by Dlag2. The test will be in fact better if you leave it set to a random value in case our implementation does reference it by mistake.